util/test/loggerTest.js

```diff
@@ -0,0 +1,268 @@
+/* eslint no-underscore-dangle: [2, { "allow": ["__get__", "_logInfo", "_logError",
+"_logDebug"] }] */
+
+/* The writing in log file is async. Hence, even after specifying a wait period,
+there is a chance the tests may fail. */
+
+const chai = require('chai');
+const winston = require('winston');
+const Logger = require('../logger.js').Logger;
+const { exec } = require('child_process');
+const fs = require('fs');
+const chaiFs = require('chai-fs');
+const path = require('path');
+const dirtyChai = require('dirty-chai');
+const rewire = require('rewire');
+const config = require('../../deploy/configs/util/logger.json');
+
+const logDirectory = config.logDirectory;
+const createLoggerObject = rewire('../logger.js').__get__('createLoggerObject');
+
+chai.should();
+chai.use(chaiFs);
+chai.use(dirtyChai);
+
+/* These variables will be used in the following tests.
+They have been declared in the global space to avoid repetition. */
+let logger;
+let secondLogger;
+const componentConfig = {
+  role: 'load_balancer',
+  hostname: 'localhost',
+  port: '8081',
+  cmd: 'log',
+};
+const secondComponentConfig = {
+  role: 'main_server',
+  hostname: 'localhost',
+  port: '9000',
+  cmd: 'log',
+};
+const logMessage = ['Error message', 'Debug message', 'Info message'];
+const logLevel = ['error', 'debug', 'info'];
+const fileNamePrefix = `${componentConfig.role}_${componentConfig.hostname}_${componentConfig.port}`;
+
+const log = logLevel.map(x => `${fileNamePrefix}_${x}.log`);
+
+const logPath = log.map(x => path.join(logDirectory, x));
+
+const deleteLogDirectory = function deleteLogDirectory(response) {
+  exec(`rm -rf ${logDirectory}`, (error) => {
+    if (error) {
+      console.error(`exec error in deleting log directory: ${error}`);
+      response(false);
+    }
+    response(true);
+  });
+};
+
+const fileContentChecker = function fileContentChecker(fileName, message, level, response) {
+  const filePath = path.join(logDirectory, fileName);
+  const fileContent = fs.readFileSync(filePath, 'utf8');
+  (JSON.parse(fileContent).message).should.equal(message);
+  (JSON.parse(fileContent).level).should.equal(level);
+  response();
+};
+
+describe('Test helper functions', () => {
```

Comment:
The describe statements need to be positive and definitive. For example, here, we can use, "Helper functions work correctly"